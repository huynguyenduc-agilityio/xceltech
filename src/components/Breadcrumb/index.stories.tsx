import type { Meta, StoryObj } from '@storybook/react';

import Breadcrumb from '.';

const defaultProps = [
  { label: 'Dashboard', href: '/dashboard' },
  { label: 'Apply for Leave' },
];

const meta: Meta<typeof Breadcrumb> = {
  title: 'Components/Breadcrumb',
  component: Breadcrumb,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],

  render: (arg) => (
    <div className="flex justify-center w-[1000px] p-9 bg-blue-light">
      <Breadcrumb {...arg} />
    </div>
  ),
} satisfies Meta<typeof Breadcrumb>;

export default meta;

type Story = StoryObj<typeof Breadcrumb>;

export const Default: Story = {
  args: { items: defaultProps },
};
