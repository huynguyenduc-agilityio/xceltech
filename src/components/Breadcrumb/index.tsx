import React from 'react';

type BreadcrumbItem = {
  label: string;
  href?: string;
};

type BreadcrumbProps = {
  items: BreadcrumbItem[];
};

const Breadcrumb = ({ items }: BreadcrumbProps) => {
  return (
    <nav className="flex items-center gap-2 text-xl text-black-soft">
      {items.map((item, index) => (
        <React.Fragment key={index}>
          {/* TODO: Convert <a> to <Link> */}
          {item.href ? (
            <a href={item.href} className="hover:underline">
              {item.label}
            </a>
          ) : (
            <span>{item.label}</span>
          )}
          {index < items.length - 1 && '>'}
        </React.Fragment>
      ))}
    </nav>
  );
};

export default Breadcrumb;
