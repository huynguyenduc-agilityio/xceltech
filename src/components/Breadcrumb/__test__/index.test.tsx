import { render } from '@testing-library/react';

// Components
import Breadcrumb from '..';

describe('Breadcrumb component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Breadcrumb
        items={[
          { label: 'Dashboard', href: '/dashboard' },
          { label: 'Apply for Leave' },
        ]}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
