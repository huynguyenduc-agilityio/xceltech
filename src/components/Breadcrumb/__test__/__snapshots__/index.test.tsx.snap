// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex items-center gap-2 text-xl text-black-soft"
  >
    <a
      class="hover:underline"
      href="/dashboard"
    >
      Dashboard
    </a>
    &gt;
    <span>
      Apply for Leave
    </span>
  </nav>
</DocumentFragment>
`;
